### ISSUE TITLE
Core Data: Normalize Entity Name Mechanism

### ISSUE THREAD

**Issue by jleandroperez** [wordpress-mobile/WordPress-iOS#4489]
We've got several approaches, spread all over, when it comes to writing entity names:
- Approach A: `@"EntityName"`
- Approach B: `NSStringFromClass([Entity class])`
- Approach C: `Entity.classnameWithoutNamespaces` (Swift classes)

Let's define a standard approach, and normalize every snippet.

**Comment by koke** [comment]
I say A ;)

Since there's no guarantee that the entity name will match the class name, I think the best way to reference it is with a string.

**Comment by jleandroperez** [comment]
@koke how abooout something like this, all over?

```
class SomeEntity : NSManagedObject
{
   public static func entityName() -> String {
      return "EntityName"
   }
}
```

**Comment by koke** [comment]
I'd be happy with that, but I still think a string is good enough 😁

**Comment by jleandroperez** [comment]
Let's get this done then. Thanks @koke! consistency FTW!

:smile:

**Comment by jleandroperez** [comment]
We've implemented CoreDataHelpers.swift which extends NSManagedObject, adding a static method called `entityName`.

However: in Unit Testing targets, the mechanism used to retrieve the entity name breaks, and we fall back to a second mechanism.

Long story short: since some classnames don't match their actual Core Data entity names, probably the best way to solve this is by means of a simple string (even if it's error prone), as discussed a couple years ago.

Closing this one for now.
